mitosheet/mitosheet/streamlit/mito_component.py

```diff
@@ -0,0 +1,123 @@
+import json
+import os
+from typing import Any, Dict, List, Callable, Optional, Tuple, Union
+
+import pandas as pd
+
+from mitosheet.mito_backend import MitoBackend
+from mitosheet.utils import get_new_id
+
+
+try:
+    import streamlit.components.v1 as components
+    import streamlit as st
+
+
+    parent_dir = os.path.dirname(os.path.abspath(__file__))
+
+    mito_build_dir = os.path.join(parent_dir, "mitoBuild")
+    _mito_component_func = components.declare_component("my_component", path=mito_build_dir)
+
+    message_passer_build_dr = os.path.join(parent_dir, "messagingBuild")
+    _message_passer_component_func = components.declare_component("message-passer", path=message_passer_build_dr)
+
+    @st.cache_resource
+    def _get_mito_backend(
+            *args: Union[pd.DataFrame, str, None], 
+            _importers: Optional[List[Callable]]=None, 
+            df_names: Optional[List[str]]=None,
+            key: Optional[str]=None
+        ) -> Tuple[MitoBackend, List[Any]]: # So it caches on key
+        mito_backend = MitoBackend(*args, user_defined_importers=_importers)
+
+        # Make a send function that stores the responses in a list
+        responses = []
+        def send(response):
+            responses.append(response)
+        
+        mito_backend.mito_send = send
+
+        if df_names is not None:
+            mito_backend.receive_message(
+                {
+                    'event': 'update_event',
+                    'id': get_new_id(),
+                    'type': 'args_update',
+                    'params': {
+                        'args': df_names
+                    },
+                }
+            )
+
+        return mito_backend, responses
+
+    def message_passer_component(key: Optional[str]=None) -> Any:
+        """
+        This component simply passes messages from the frontend to the backend,
+        so that the backend can process them before it is rendered.
+        """
+        component_value = _message_passer_component_func(key=key)
+        return component_value
+
+
+    def mito_component( # type: ignore
+            *args: Union[pd.DataFrame, str, None], 
+            importers: Optional[List[Callable]]=None, 
+            df_names: Optional[List[str]]=None,
+            key=None
```

Comment:
I briefly tested that if I have two different keys than I can have two mitosheets in the same dashboard and it was working